include/roboteam_ai/control/positionControl/pathPlanning/VoronoiPathPlanning.h

```diff
@@ -0,0 +1,81 @@
+//;
+// Created by ratoone on 05-11-19.
+//
+
+#ifndef RTT_VORONOIPATHPLANNING_H
+#define RTT_VORONOIPATHPLANNING_H
+
+#include "Voronoi.h"
+#include <queue>
+#include <list>
+#include <roboteam_utils/Vector2.h>
+#include <utilities/Constants.h>
+#include <control/ControlUtils.h>
+#include "control/positionControl/CollisionDetector.h"
+
+namespace rtt::ai::control{
+struct GraphNode{
+    rtt::Vector2 nextNodePosition;
+    double distance;
+};
+
+//black magic vector hashing - taken from the boost hash combine
+struct hashPoint{
+    size_t operator()(const rtt::Vector2& point) const {
+        size_t xHash = std::hash<double>()(point.x) + 0x9e3779b9;
+        return xHash ^ (std::hash<double>()(point.y) + 0x9e3779b9 + (xHash<<6) + (xHash>>2));
+    }
+};
```

Comment:
How relevant is this for speed?
If not necessary I would avoid it.

Reply:
Vector2 naturally isn't hashable so you kind of have to hash it to be able to store it in a hashmap

Reply:
We might want to add this to roboteam_utils then for some types. 
It's nice if it's defined